xbmc/cores/VideoPlayer/DVDCodecs/Audio/DVDAudioCodecPassthrough.cpp

```diff
@@ -127,12 +132,25 @@ int CDVDAudioCodecPassthrough::Decode(uint8_t* pData, int iSize, double dts, dou
     m_bufferSize = std::max(m_bufferSize, m_dataSize);
     if (consumed != m_backlogSize)
     {
-      memmove(m_backlogBuffer, m_backlogBuffer+consumed, consumed);
+      memmove(m_backlogBuffer, m_backlogBuffer+consumed, m_backlogSize-consumed);
       m_backlogSize -= consumed;
     }
   }
 
-  if (pData && !m_dataSize)
+  // get rid of potential side data
+  if (pData)
+  {
+    AVPacket pkt;
+    av_init_packet(&pkt);
+    pkt.data = pData;
+    pkt.size = iSize;
+    av_packet_split_side_data(&pkt);
+    skip = iSize - pkt.size;
+    pData = pkt.data;
+    iSize = pkt.size;
```

Comment:
I think you miss an av_packet_free_side_data here, as av_packet_split_side_data mallocs a buffer to copy the sidedata?